tests/components/metoffice/test_config_flow.py

```diff
@@ -0,0 +1,140 @@
+"""Test the National Weather Service (NWS) config flow."""
+import json
+
+from asynctest import patch
+
+from homeassistant import config_entries, setup
+from homeassistant.components.metoffice.const import DOMAIN
+
+from .const import (
+    TEST_API_KEY,
+    TEST_LATITUDE_WAVERTREE,
+    TEST_LONGITUDE_WAVERTREE,
+    TEST_SITE_NAME_WAVERTREE,
+)
+
+from tests.common import load_fixture
+
+
+async def test_form(hass, requests_mock):
+    """Test we get the form."""
+    hass.config.latitude = TEST_LATITUDE_WAVERTREE
+    hass.config.longitude = TEST_LONGITUDE_WAVERTREE
+
+    # all metoffice test data encapsulated in here
+    mock_json = json.loads(load_fixture("metoffice.json"))
+    all_sites = json.dumps(mock_json["all_sites"])
+    requests_mock.get("/public/data/val/wxfcs/all/json/sitelist/", text=all_sites)
+
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == "form"
+    assert result["errors"] == {}
+
+    with patch(
+        "homeassistant.components.metoffice.async_setup", return_value=True
+    ) as mock_setup, patch(
+        "homeassistant.components.metoffice.async_setup_entry", return_value=True,
+    ) as mock_setup_entry:
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"], {"api_key": TEST_API_KEY}
+        )
+
+    assert result2["type"] == "create_entry"
+    assert result2["title"] == TEST_SITE_NAME_WAVERTREE
+    assert result2["data"] == {
+        "api_key": TEST_API_KEY,
+        "latitude": TEST_LATITUDE_WAVERTREE,
+        "longitude": TEST_LONGITUDE_WAVERTREE,
+        "name": TEST_SITE_NAME_WAVERTREE,
+    }
+    await hass.async_block_till_done()
+    assert len(mock_setup.mock_calls) == 1
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_form_already_configured(hass, requests_mock):
+    """Test we handle duplicate entries."""
+    hass.config.latitude = TEST_LATITUDE_WAVERTREE
+    hass.config.longitude = TEST_LONGITUDE_WAVERTREE
+
+    # all metoffice test data encapsulated in here
+    mock_json = json.loads(load_fixture("metoffice.json"))
+
+    all_sites = json.dumps(mock_json["all_sites"])
+    requests_mock.get("/public/data/val/wxfcs/all/json/sitelist/", text=all_sites)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch(
+        "homeassistant.components.metoffice.async_setup", return_value=True
+    ) as mock_setup, patch(
+        "homeassistant.components.metoffice.async_setup_entry", return_value=True,
+    ) as mock_setup_entry:
+        result2 = await hass.config_entries.flow.async_configure(
```

Comment:
Just add a MockConfigEntry for the first entry instead.
https://github.com/home-assistant/core/blob/5f8dcd45c1a4279f792f39ef6deffb8f8677cef5/tests/common.py#L668

Reply:
Think I've done it

Reply:
Nope. I've plainly misunderstood. Could you explain how I should be using the MockConfigEntry, please? I've used one in place of the first entry but it's obviously not what  I was supposed to be doing as now the second entry gets through to "create_entry" instead of "abort"ing?

Reply:
Think I'm there - just having fun and game commiting - something about .pre-commit-hooks.yaml being missing. I'm going to commit straight into GitHub, though, as I've tested it locally.